src/array/cpu/spmm.h

```diff
@@ -42,6 +42,8 @@ template <typename IdType, typename DType, typename Op>
 void SpMMSumCsrNaive(
     const BcastOff& bcast, const CSRMatrix& csr, const DType* X, const DType* W,
     DType* O) {
+  typedef typename std::conditional<
+      std::is_same<DType, BFloat16>::value, float, DType>::type AccType;
```

Comment:
Why not make it a file-wide typedef?

Reply:
It depends on DType and appears only in two Naive methods. I don't have good idea right now how to move it in pretty way.

Reply:
Try 
```
template <typename DType>
using AccType = typename std::conditional<std::is_same<DType, BFloat16>::value, float, DType>::type;
```

It should work with c++ 11.

Reply:
thanks, changed